pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -125,14 +163,128 @@ func localityLbEndpointsFromInstances(instances []*model.ServiceInstance) []endp
 			}
 			localityEpMap[locality] = locLbEps
 		}
-		log.Infoa("EDS LOCALITY ep: ", (*lbEp).String(), localityEpMap)
 		locLbEps.LbEndpoints = append(locLbEps.LbEndpoints, *lbEp)
 	}
-	log.Infoa("EDS LOCALITY: ", localityEpMap)
 	out := make([]endpoint.LocalityLbEndpoints, 0, len(localityEpMap))
 	for _, locLbEps := range localityEpMap {
 		out = append(out, *locLbEps)
-		log.Infoa("EDS LOCALITY add: ", locLbEps.String())
 	}
 	return out
 }
+
+// StreamEndpoints implements xdsapi.EndpointDiscoveryServiceServer.StreamEndpoints().
+func (s *DiscoveryServer) StreamEndpoints(stream xdsapi.EndpointDiscoveryService_StreamEndpointsServer) error {
+	ticker := time.NewTicker(responseTickDuration)
+	peerInfo, ok := peer.FromContext(stream.Context())
+	peerAddr := "Unknown peer address"
+	if ok {
+		peerAddr = peerInfo.Addr.String()
+	}
+	defer ticker.Stop()
+	var discReq *xdsapi.DiscoveryRequest
+	var receiveError error
+	reqChannel := make(chan *xdsapi.DiscoveryRequest, 1)
+	con := & EdsConnection {
+		pushChannel : make(chan bool, 1),
+		peerAddr: peerAddr,
+		clusters: []string{},
+	}
+	go func() {
```

Comment:
suggest moving this to helper function (if this is not throwaway code). Function is getting a bit long to read easily.